File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        
        for(int i=0; i<n; i++){
            if(!visited[i]){
                dfs(i, computers, n, visited);
                answer++;         
            }
        }
        
        return answer;
    }
    void dfs(int index,int[][] computers, int n, boolean[] visited){
        visited[index] = true;
        for(int i=0; i<n; i++){
            if(computers[index][i]==1 && !visited[i]){
                dfs(i, computers, n, visited);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    
    const visited = new Array(n).fill(false);
    // console.log(visited);
    
    for(let i=0; i<n; i++){
        if(!visited[i]){
            dfs(visited, computers, n, i);
            answer++;
        }
    }
    
    return answer;
}

function dfs(visited, computers, n, index){
    visited[index]=true;
    for(let i=0; i<n; i++){
        if(computers[index][i]==1 && !visited[i]){
            dfs(visited, computers, n, i);
        }
    }
}